keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,403 @@
+it would also be possible to make it a separate top-level config API or 
+use a config map containing a single yaml field that would be deserialized
+into the type defined below.
+
+```go
+type ResourceTransformationStrategy string
+const Retain ResourceTransformationStrategy = "Retain"
+const Replace ResourceTransformationStrategy = "Replace"
+
+type ResourceTransformation struct {
+  // key is the output resource name and value is the quantity of the output resource 
+  // that corresponds to 1 unit of the input resource
+  Functions map[corev1.ResourceName ]resource.Quantity `json:"functions"`
+
+  // should the input resource be replaced or retained
+  Strategy ResourceTransformationStrategy `json:"strategy"`
+}
+
+type ResourceTransformationMatrix struct {
+  // The transformations to apply where the map key is the input resource name
+  Transformations map[corev1.ResourceName]ResourceTransformation `json:"transformations"`
+}
+```
+A key advantage of defining a single object is that it gives a cluster admin
+a single source of truth.  By using maps instead of arrays, we avoid needing
+to define the semantics of multiple entries having the same input or output
+resource name.  Using the input resource name as the key to the outer map
+avoids conflicting definitions of `Replace` for an input resource.
+
+### Option 2: Singleton Matrix Output Focused
+This is similar to Option 1, but makes the output resource the
+primary key.  
+```go
+type ResourceTransformationStrategy string
+const Retain ResourceTransformationStrategy = "Retain"
+const Replace ResourceTransformationStrategy = "Replace"
+
+type ResourceTransformationMatrix struct {
+  // The transformations to apply where the first map key is the output resource name
+  // and the inner map key is the input resource name
+  Transformations map[corev1.ResourceName]map[corev1.ResourceName]resource.Quantity `json:"transformations"`
+
+  // The map key is the input resource name.
+  Strategy map[corev1.ResourceName]ResourceTransformationStrategy `json:"strategy"`
+}
+```
+
+Grouping by output makes it easier for the cluster admin to see/specify
+all the resources that map into a given output as a unit.  However it
+comes at the cost of a separate map to specify strategy.
+It may be confusing that the `Transformations` map is keyed by
+output name as the primary key while `Strategy` is keyed by input name.
+
+### Comparison
+
+Consider the following list of 4-tuples
+
+* (`nvidia.com/mig-1g.5gb`, `kueue.x-k8s.io/accelerator-memory`, `5G`, `replace`)
+* (`nvidia.com/mig-2g.10gb`, `kueue.x-k8s.io/accelerator-memory`, `10G`, `replace`)
+* (`nvidia.com/mig-4g.20gb`, `kueue.x-k8s.io/accelerator-memory`, `20G`, `replace`)
+* (`nvidia.com/mig-1g.5gb`, `kueue.x-k8s.io/credits`, `10`, `replace`)
+* (`nvidia.com/mig-2g.10gb`, `kueue.x-k8s.io/credits`, `15`, `replace`)
+* (`nvidia.com/mig-4g.20gb`, `kueue.x-k8s.io/credits`, `30`, `replace`)
+* (`cpu`, `kueue.x-k8s.io/credits`, `1`, `retain`)
+
+Option 1
```

Comment:
I'm leaning towards option 1 - yes, it is more verbose, but seems more intuitive, with less hidden assumptions.
I think it will also be easier to evolve in the future (if ever) if we want to introduce more flexible transformation functions.